geomstats/geometry/product_manifold.py

```diff
@@ -311,3 +318,291 @@ def is_tangent(self, vector, base_point, atol=gs.atol):
 
         is_tangent = gs.all(is_tangent, axis=-1)
         return is_tangent
+
+
+class NFoldManifold(Manifold):
```

Comment:
Maybe it should have its own `nfold_manifold.py` file, for visibility for new users?

Reply:
It does not inherit from ProductManifold bcs the machinery is different?

Reply:
Yes ProductManifold won't work for matrix point types, so I started from scratch to avoid this issue.
I think a user might be looking for a product manifold before realizing they can use the nfold so it's good to keep it here. The name of the file won't matter though if we get to do imports from the geometry module itself. Did you try?

Reply:
Makes sense.

I feel that exposing a module `nfold_manifold.py` makes it easier for the user going through the files (on GitHub) to see that we have it. But that may be personal, and it does not matter too much.

Yes I tried doing the imports, it was working first but then I ran into circular imports, and stopped: I will not have the bandwidth to do it this quarter. There is a closed PR on this, if you want to have a look. I wonder if having one class per .py could help disentangle the imports?